src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,44 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.exception.TaskIndexOutOfBoundException;
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * A class represents a DeleteCommand.
+ */
+public class DeleteCommand extends IndexCommand {
+    /**
+     * Constructs a DeleteCommand.
+     * @param fullCommand Full command from the user's input.
+     */
+    public DeleteCommand(String fullCommand) {
+        super(fullCommand);
+    }
+
+    /**
+     * Deletes the specified task from the task list.
+     * @param tasks Task list given.
+     * @param ui User interface class object.
+     * @param storage Storage path that is going to be updated.
+     * @throws DukeException If error occurs during the process.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        int index = getIndex();
+        if (index < tasks.size()) {
+            Task removingTask = tasks.get(index);
+            tasks.remove(index);
+            storage.updateInFile(tasks);
+            System.out.println(" Following task is removed:");
+            System.out.println("  " + removingTask);
+            System.out.println(" Now you have " + tasks.size() + " tasks.");
```

Comment:
You have a `Ui` object passed to the command, which ideally handles message passing to `System.out`.
Can consider offloading these to the `ui` instead.

Same as in the other commands.

Reply:
I also realised this and I will make some modification later. Thanks for pointing it out!